mesh/mesh.cpp

```diff
@@ -1552,6 +1552,153 @@ Geometry::Type Mesh::GetTypicalElementGeometry() const
    return geom;
 }
 
+void Mesh::GetExteriorFaceMarker(Array<int> & face_marker) const
+{
+   if (face_marker.Size() < faces_info.Size())
+   {
+      face_marker.SetSize(faces_info.Size());
+   }
+
+   for (int f = 0; f < faces_info.Size(); f++)
+   {
+      if (faces_info[f].NCFace == -1 &&
+          faces_info[f].Elem2No < 0 &&
+          faces_info[f].Elem2Inf < 0 )
+      {
+         face_marker[f] = 1;
+      }
+      else
+      {
+         face_marker[f] = 0;
+      }
+   }
+}
+
+void Mesh::UnmarkInternalBoundaries(Array<int> &bdr_marker, bool excl) const
+{
+   MFEM_VERIFY(bdr_marker.Size() >= bdr_attributes.Max(),
+               "bdr_marker must be at least bdr_attriburtes.Max() in length");
+
+   Array<bool> interior_bdr(bdr_attributes.Max()); interior_bdr = false;
+   Array<bool> exterior_bdr(bdr_attributes.Max()); exterior_bdr = false;
+
+   // Identify attributes which contain interior faces and those which
+   // contain exterior faces.
+   for (int be = 0; be < boundary.Size(); be++)
+   {
+      const int bea = boundary[be]->GetAttribute();
+
+      if (bdr_marker[bea-1] != 0)
+      {
+         const int f = be_to_face[be];
+
+         if (faces_info[f].NCFace == -1 &&
+             faces_info[f].Elem2No < 0 &&
+             faces_info[f].Elem2Inf < 0 )
+         {
+            exterior_bdr[bea-1] = true;
+         }
+         else
+         {
+            interior_bdr[bea-1] = true;
+         }
+      }
+   }
+
+   // Unmark attributes which are currently marked, contain interior faces,
+   // and satisfy the appropriate exclusivity requirement.
+   for (int b = 0; b < bdr_attributes.Max(); b++)
```

Comment:
`bdr_attributes.Max()` is not O(1) method, so it is best to precompute and store this value.

This change applies to other newly added methods too.

Reply:
Good point. Done